src/solvespace.h

```diff
@@ -229,7 +230,7 @@ class System {
 
     // A list of parameters that are being dragged; these are the ones that
     // we should put as close as possible to their initial positions.
-    List<hParam>                    dragged;
+    ParamSet                        dragged;
```

Comment:
Please, carefully test it. For small dragging lists std::vector is preferrable, but depending on size of the list, custom container can be created (using linear search for small amount of data and switch to hastable when data is huge). The point of using such things - you can drag single point of huge object and it is more often way to drag, but also you can drag a big number of objects, so it is worth to optimize both cases.

Reply:
This is again more of a cleanup than an optimisation. I did test it, and while there can be a potential gain with an extremely large amount of dragged parameters, in practice it doesn't make a meaningful difference in the grand scheme of things the way SolveSpace uses it, especially given the fact it's only used when solving by substitution and scaling the parameters, while when solving the bulk of the time is spent inside Eigen when solving and rank testing anyway.